sdk/storage/azure-storage-file-datalake/azure/storage/file/out/datalake/_path_client.pyi

```diff
@@ -0,0 +1,18 @@
+# Stubs for datalake._path_client (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from ._data_lake_lease import DataLakeLeaseClient
+from ._generated import DataLakeStorageClient
+from azure.storage.blob._shared.base_client import StorageAccountHostsMixin
+from typing import Any, Optional
+
+class PathClient(StorageAccountHostsMixin):
```

Comment:
I would very, very much like to merge the `PathClient`, `DataLakeDirectoryClient` and `DataLakeFileClient` into a single class.

Reply:
Is it to merge DatalakeDirectoryClient and DataLakeFileClient into their current parent PathClient?
Then PathClient will have mixed operation for both file and directory. 

eg. PathClient will have:
- get_file_properties and get_directory_properties
- create_directory and create_file
- append_data, flush_data, read_file which are only for files

Reply:
Yes. And understood.

Reply:
And I would add `is_file()` and `is_directory()` methods to the common `PathClient`.

Reply:
further discussion